### PR TITLE
Add locked, locked_by, locked_on fields to project endpoint

### PR DESCRIPTION
**Pull request by rashidg**
Refs #SDE-2576

### CODE REVIEW

File: source/includes/projects.md [hunk 1]
```diff
@@ -45,7 +45,10 @@ Content-Type: application/json
         }],
         "custom_attributes": {
           "slug": "value"
-        }
+        },
+        "locked_on": null,
+        "locked_by": null,
```
Comment (geoffwhittington) [inline source/includes/projects.md]: Is this a user item_id or a user id? Can API users get the full user details (name, email, etc)

File: source/includes/projects.md [hunk 2]
```diff
@@ -142,7 +145,10 @@ Content-Type: application/json
                 "complete": 1,
                 "na": 0
             }
-        }
+        },
+        "locked_on": null,
+        "locked_by": null,
+        "locked": false
```
Comment (geoffwhittington) [inline source/includes/projects.md]: Can you provide a real example?

File: source/includes/projects.md [hunk 3]
```diff
@@ -293,7 +305,7 @@ groups      | No       | A list of dictionaries per group that are to be assigne
 name        | Yes      | The name of the new project.
 profile     | No       | The ID of the desired profile for the project.
 users       | No       | A list of dictionaries per user that are to be assigned to the project. Each dictionary should contain the user's email and the desired role.
-
+locked      | No       | A boolean field to lock or unlock the project. It can only be used by users that have lock_project_settings permission
```
Comment (geoffwhittington) [inline source/includes/projects.md]: Can API users PATCH the "locked" attribute only?

Comment (RouganStriker) [inline source/includes/projects.md]: Yep

File: source/includes/projects.md (hunk 1 above)
Comment (RouganStriker) [inline source/includes/projects.md]: I believe locked_by displays the user's email

Comment (geoffwhittington) [inline source/includes/projects.md]: We should consider making this a user item_id. We're inconsistent in the API - but item_id's are the way forward as far as I know. Correct @funkaoshi ?

Comment (funkaoshi) [inline source/includes/projects.md]: Yes, and we should make them expandable, ideally.

Comment (RouganStriker) [inline source/includes/projects.md]: Oops I was wrong, it displays the user id. It is currently not scoped but we can change that field to display the item_id and include an expand filter.

File: source/includes/projects.md (hunk 2 above)
Comment (RouganStriker) [inline source/includes/projects.md]: Updated the PUT example

**Pull request merged**